File: multicontactpicker/src/main/java/com/wafflecopter/multicontactpicker/RxContacts/PhoneNumber.java
package com.wafflecopter.multicontactpicker.RxContacts;

import android.os.Parcel;
import android.os.Parcelable;

public class PhoneNumber implements Parcelable {

    private String typeLabel;
    private String number;

    PhoneNumber(String typeLabel, String number) {
        this.typeLabel = typeLabel;
        this.number = number;
    }

    private PhoneNumber(Parcel in) {
        this.typeLabel = in.readString();
        this.number = in.readString();
    }

    public String getTypeLabel() {
        return typeLabel;
    }

    public void setTypeLabel(String typeLabel) {
        this.typeLabel = typeLabel;
    }

    public String getNumber() {
        return number;
    }

    public void setNumber(String number) {
        this.number = number;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.typeLabel);
        dest.writeString(this.number);
    }

    public static final Creator<PhoneNumber> CREATOR = new Creator<PhoneNumber>() {
        @Override
        public PhoneNumber createFromParcel(Parcel in) {
            return new PhoneNumber(in);
        }

        @Override
        public PhoneNumber[] newArray(int size) {
            return new PhoneNumber[size];
        }
    };
}

File: contactspicker/src/main/java/com/wafflecopter/contactspicker/contactsmodel/PhoneNumber.kt
package com.wafflecopter.contactspicker.contactsmodel

import android.os.Parcel
import android.os.Parcelable

class PhoneNumber : Parcelable {
    var typeLabel: String?
    var number: String?

    internal constructor(typeLabel: String?, number: String?) {
        this.typeLabel = typeLabel
        this.number = number
    }

    private constructor(`in`: Parcel) {
        typeLabel = `in`.readString()
        number = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(typeLabel)
        dest.writeString(number)
    }

    companion object {
        var CREATOR: Parcelable.Creator<PhoneNumber> = object : Parcelable.Creator<PhoneNumber> {
            override fun createFromParcel(`in`: Parcel): PhoneNumber? {
                return PhoneNumber(`in`)
            }

            override fun newArray(size: Int): Array<PhoneNumber?> {
                return arrayOfNulls(size)
            }
        }
    }
}